source/3dcitydb/docker.rst

```diff
@@ -460,44 +460,108 @@ that may help to increase database performance.
+lecture or workshop, that will run out of the box, without having to import
+data first.
 
-In general, it is not recommended to store data directly inside a Docker image...
+.. warning:: The practise described here has many drawbacks and is a potential
+  security threat. It should not be performed with sensitive data!
 
+Here is how to create an image with data:
 
-1. Start a 3DCityDB Docker container.
+1. Choose a 3DCityDB image that is suitable for you purpose. You will not be able
+   to change the image version later, as you could easily do when using volumes
+   (the default). Available versions are listed in :ref:`citydb_docker_image_variants`.
+   To update an image with data, it has to be recreated from scrap using the
+   desired/updated base image.
+
+2. Start a 3DCityDB Docker container:
 
   .. code-block:: bash
 
-    docker run -d --name citydb \
+    docker run -d --name citydbTemp \
       -e "PGDATA=/mydata" \
       -e "POSTGRES_PASSWORD=changeMe!" \
       -e "SRID=25832" \
     3dcitydb/3dcitydb-pg:4.1.0-alpine
 
-2. Import data to the container. For this example we are use the
+  .. warning:: The database credentials and settings provided in this step
+    cannot be changed when later on creating containers from this image!
+
+    Note down the database connection credentials (db name, username, password)
+    or you won't be able to access the content later.
+
+2. Import data to the container. For this example we are using the
    `LoD3 Railway dataset <https://github.com/3dcitydb/importer-exporter/raw/
    92e08aa306611ee850e065bb542bb3d60791a54f/resources/samples/
-   Railway%20Scene/Railway_Scene_LoD3.zip>`_.
+   Railway%20Scene/Railway_Scene_LoD3.zip>`_ and the
+   :ref:`3DCityDB Importer/Exporter Docker image<impexp_docker_chapter>`:
 
   .. code-block:: bash
 
     docker run -i -t --rm --name impexp \
-        --link citydb \
+        --link citydbTemp \
         -v /d/temp:/data \
       3dcitydb/impexp:edge-alpine import \
-        -H citydb \
+        -H citydbTemp \
         -d postgres \
         -u postgres \
         -p changeMe! \
-        /data/building.gml
+        /data/Railway_Scene_LoD3.zip
+
+3. Stop the running 3DCityDB container and commit it to an image:
+
+  .. code-block:: bash
 
-3. Stop running 3DCityDB container commit 3DCityDB container to an image
+    docker stop citydbTemp
+    docker commit citydbTemp 3dcitydb/3dcitydb-pg:4.1.0-alpine-railwayScene_LoD3
+
+4. Remove the 3DCityDB container:
 
   .. code-block:: bash
 
-    docker stop citydb
-    docker commit citydb 3dcitydb/3dcitydb-pg:4.1.0-alpine-railwayScene_LoD3
+    docker rm -f -v citydbTemp
+
+We have now created a 3DCityDB image that contains data that can e.g. be pushed to a
+Docker registry or exported ar TAR.
+When creating containers from this image, it is not required to specify any configuration
+parameter as you usually would, when creating a fresh 3DCityDB container.
```

Comment:
I am new to this. What kind of configuration parameters are not required anymore?

Reply:
The PostgreSQL image checks on startup weather a database already exists or not. If there is one, the initialisation that parses the ENV vars and creates the default DB in the container is skipped. Hence, setting a db password, user usw. ist not possible. That means with images, that already contain data (a database), these parameters can be omitted.

That is acually quite cool for testing images. As the database initialisation and data import is not required, you have a running database you can connect to directly (really instantly) after container starts.